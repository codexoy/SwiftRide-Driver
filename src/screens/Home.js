import React from "react"
import { View, Text, Button, StyleSheet } from "react-native"
import { logout } from "../actions/auth"
import { connect } from "react-redux"

class Home extends React.Component {
	constructor(props) {
		super(props)
		this.logout = this.logout.bind(this)
	}

	logout() {
		this.props.dispatch(
			logout(() => {
				this.props.navigation.navigate("Auth")
			})
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<Text>Hi!</Text>
				<Button onPress={this.logout} title="התנתק" />
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})

export default connect()(Home)
