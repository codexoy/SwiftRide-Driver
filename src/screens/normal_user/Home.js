import React from "react"
import { View, StyleSheet } from "react-native"
import { logout } from "../../actions/auth"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import SuccessModal from "../../components/SuccessModal"
import { signUpRoles } from "../../consts"
import { deleteDeviceToken } from "../../actions/utils"

export class Home extends React.Component {
	constructor(props) {
		super(props)
		this.logout = this.logout.bind(this)
		this.role = "normal"
		if (this.props.user.hasOwnProperty("teacher_id")) {
			// it's a teacher
			this.role = signUpRoles.teacher
		} else if (this.props.user.hasOwnProperty("my_teacher")) {
			// it's a student
			this.role = signUpRoles.student
		}
	}

	async logout() {
		await this.props.dispatch(deleteDeviceToken())
		this.props.dispatch(
			logout(async () => {
				this.props.navigation.navigate("Auth")
			})
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<SuccessModal
					animationType="none"
					visible={true}
					image="pending"
					title={strings("normal_user.home.pending")}
					desc={strings(
						"normal_user.home." + this.role + "_pending_desc"
					)}
					buttonPress={() => {
						this.logout()
					}}
					button={strings("settings.logout")}
				/>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})
const mapStateToProps = state => {
	return {
		user: state.user
	}
}

export default connect(mapStateToProps)(Home)
