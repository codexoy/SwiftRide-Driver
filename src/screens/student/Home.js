import React, { Fragment } from "react"
import {
	ScrollView,
	View,
	Text,
	Image,
	StyleSheet,
	TouchableHighlight,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import ShadowRect from "../../components/ShadowRect"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import LessonPopup from "../../components/LessonPopup"
import { getPayments } from "../../actions/lessons"
import { MAIN_PADDING } from "../../consts"
import StudentPayments from "../../components/StudentPayments"
import StudentNextLessonView from "../../components/StudentNextLessonView"

export class Home extends React.Component {
	static navigationOptions = () => {
		return {
			title: "home",
			tabBarLabel: strings("tabs.home"),
			tabBarAccessibilityLabel: strings("tabs.home"),
			tabBarTestID: "HomeTab"
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			lesson: "",
			payments: [],
			lessonPopupVisible: false,
			loading: true
		}

		this._handleRequests()
	}

	_handleRequests = async () => {
		await this._getLesson()
		await this._getPayments()
		this.setState({ loading: false })
	}
	_navigateToSettings = () => {
		this.props.navigation.navigate("Settings")
	}

	_getLesson = async () => {
		const now = new Date().toISOString()
		const resp = await this.props.fetchService.fetch(
			"/lessons/?limit=1&is_approved=true&date=ge:" + now,
			{ method: "GET" }
		)
		if (resp.json["data"].length == 0) return
		this.setState({
			lesson: resp.json["data"][0]
		})
	}

	_getPayments = async () => {
		const payments = await getPayments(this.props.fetchService)
		this.setState({
			payments: payments.payments
		})
	}

	lessonPress = () => {
		this.setState({ lessonPopupVisible: !this.state.lessonPopupVisible })
	}

	renderLesson = () => {
		const { lesson } = this.state
		return (
			<Fragment>
				<StudentNextLessonView
					onPress={() => this.lessonPress()}
					testID="lessonRowTouchable"
					lesson={lesson}
					loading={this.state.loading}
				/>
				<LessonPopup
					visible={this.state.lessonPopupVisible}
					item={lesson}
					onPress={this.lessonPress}
					testID="lessonPopup"
					navigation={this.props.navigation}
				/>
			</Fragment>
		)
	}

	render() {
		return (
			<ScrollView>
				<View style={styles.container}>
					<View style={styles.settingsIcon}>
						<TouchableOpacity
							onPress={this._navigateToSettings.bind(this)}
						>
							<Icon name="settings" type="material" size={24} />
						</TouchableOpacity>
					</View>
					<View testID="welcomeHeader" style={styles.welcomeHeader}>
						<Image
							style={styles.profilePic}
							source={{
								uri:
									"https://images.unsplash.com/photo-1535643302794-19c3804b874b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2134&q=80"
							}}
						/>
						<Text style={styles.welcomeText}>
							{strings("teacher.home.welcome", {
								name: this.props.user["name"]
							})}
						</Text>
					</View>
					<ShadowRect style={styles.schedule}>
						<Text style={styles.rectTitle} testID="schedule">
							{strings("teacher.home.next_lesson")}
						</Text>
						{this.renderLesson()}
					</ShadowRect>

					<View style={styles.fullScheduleView}>
						<TouchableHighlight
							underlayColor="lightgray"
							onPress={() => {
								this.props.navigation.navigate("Schedule")
							}}
						>
							<Text style={styles.fullSchedule}>
								{strings("teacher.home.full_schedule")}
							</Text>
						</TouchableHighlight>
						<Icon
							size={20}
							color="rgb(12, 116, 244)"
							name="ios-arrow-dropleft-circle"
							type="ionicon"
						/>
					</View>
					<ShadowRect>
						<View style={{ flex: 1, flexDirection: "row" }}>
							<Text
								testID="monthlyAmount"
								style={styles.rectTitle}
							>
								{strings("student.home.payments")}
							</Text>
							<View
								style={{
									flex: 1,
									alignItems: "flex-end",
									marginRight: "auto"
								}}
							>
								<Icon
									name="arrow-back"
									type="material"
									size={20}
								/>
							</View>
						</View>
						<StudentPayments
							sum={this.props.user.balance}
							payments={this.state.payments.slice(0, 2)}
							loading={this.state.loading}
						/>
					</ShadowRect>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: 20,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	settingsIcon: {
		position: "absolute",
		right: MAIN_PADDING
	},
	profilePic: {
		width: 44,
		height: 44,
		borderRadius: 22,
		marginBottom: 16
	},
	welcomeHeader: {
		alignSelf: "center",
		alignItems: "center",
		marginBottom: 20
	},
	welcomeText: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	hoursStyle: { marginBottom: -8 },
	rectTitle: {
		fontSize: 16,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start"
	},
	lessonRow: {
		marginTop: 12
	},
	fullScheduleView: {
		flexDirection: "row",
		flex: 1,
		alignSelf: "center",
		alignItems: "center",
		marginTop: 4,
		marginBottom: 24
	},
	fullSchedule: {
		color: "rgb(12, 116, 244)",
		marginTop: -4,
		fontWeight: "bold",
		marginRight: 8
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Home)
