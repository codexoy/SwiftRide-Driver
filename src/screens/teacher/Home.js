import React, { Fragment } from "react"
import {
	ScrollView,
	View,
	Text,
	StyleSheet,
	TouchableHighlight,
	TouchableOpacity,
	FlatList
} from "react-native"
import { connect } from "react-redux"
import ShadowRect from "../../components/ShadowRect"
import UserWithPic from "../../components/UserWithPic"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import Separator from "../../components/Separator"
import { Icon } from "react-native-elements"
import Hours from "../../components/Hours"
import LessonPopup from "../../components/LessonPopup"
import { MAIN_PADDING, colors, NAME_LENGTH } from "../../consts"
import { getPayments } from "../../actions/lessons"
import EmptyState from "../../components/EmptyState"
import LessonsLoader from "../../components/LessonsLoader"
import PaymentsLoader from "../../components/PaymentsLoader"
import { NavigationActions } from "react-navigation"
import { getUserImage, uploadUserImage } from "../../actions/utils"
import UploadProfileImage from "../../components/UploadProfileImage"
import moment from "moment"

export class Home extends React.Component {
	static navigationOptions = () => {
		return {
			title: "home",
			tabBarLabel: strings("tabs.home"),
			tabBarAccessibilityLabel: strings("tabs.home"),
			tabBarTestID: "HomeTab"
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			currentLesson: null,
			nextLesson: null,
			payments: [],
			sum: 0,
			visible: [],
			loading: true
		}
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._sendRequests()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	_sendRequests = async () => {
		await this._getLessons()
		await this._getPayments()
		this.setState({ loading: false })
	}

	_getLessons = async () => {
		const now = new Date().toISOString()
		const endOfDay = moment
			.utc()
			.endOf("day")
			.toISOString()
		const currentLessonResp = await this.props.fetchService.fetch(
			"/lessons/?limit=1&is_approved=true&date=ge:" +
				now +
				"&date=le:" +
				endOfDay,
			{ method: "GET" }
		)
		const nextLessonResp = await this.props.fetchService.fetch(
			"/lessons/?limit=2&is_approved=true&date=ge:" + now,
			{ method: "GET" }
		)
		let nextLesson = null,
			currentLesson = null

		if (nextLessonResp.json["data"].length > 0) {
			if (currentLessonResp.json["data"].length == 0) {
				// no current lesson (no lessons today)
				nextLesson = nextLessonResp.json["data"][0]
			} else {
				currentLesson = currentLessonResp.json["data"][0]
				nextLesson = nextLessonResp.json["data"][1]
			}
		}

		this.setState({
			currentLesson,
			nextLesson
		})
	}

	_getPayments = async () => {
		const payments = await getPayments(this.props.fetchService)
		this.setState({
			payments: payments.payments,
			sum: payments.sum
		})
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderLesson = item => {
		if (!item) {
			return (
				<Text
					style={{
						...styles.lessonRow,
						...styles.noLesson
					}}
				>
					{strings("teacher.home.no_lesson")}
				</Text>
			)
		}
		let student = strings("teacher.no_student_applied")
		let user = null
		if (item.student) {
			student = item.student.user.name
			user = item.student.user
		}
		const date = item.date
		let meetup = strings("not_set")
		if (item.meetup_place) meetup = item.meetup_place.name
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity
					onPress={() => this.lessonPress(item)}
					testID="lessonRowTouchable"
				>
					<Row
						key={`item${item.id}`}
						style={styles.lessonRow}
						leftSide={
							<Hours
								style={styles.hours}
								duration={item.duration}
								date={date}
							/>
						}
					>
						<UserWithPic
							name={student}
							user={user}
							extra={
								<View style={{ alignItems: "flex-start" }}>
									<Text style={styles.places}>
										{strings("teacher.new_lesson.meetup")}:{" "}
										{meetup}
									</Text>
								</View>
							}
							nameStyle={styles.nameStyle}
							imageContainerStyle={styles.placesImage}
						/>
					</Row>
				</TouchableOpacity>
				<LessonPopup
					visible={visible}
					item={item}
					onPress={this.lessonPress}
					testID="lessonPopup"
					navigation={this.props.navigation}
				/>
			</Fragment>
		)
	}

	renderPaymentItem = ({ item, index }) => {
		let firstItemStyles = {}
		if (index == 0) {
			firstItemStyles = { marginTop: 0 }
		}
		return (
			<Row
				style={{ ...styles.paymentRow, ...firstItemStyles }}
				leftSide={
					<Text style={styles.amountOfStudent}>{item.amount}₪</Text>
				}
				key={`payment${item.id}`}
			>
				<UserWithPic
					user={item.student.user}
					nameStyle={styles.nameStyle}
				/>
			</Row>
		)
	}

	_renderEmpty = type => (
		<EmptyState
			image={type}
			text={strings(`empty_${type}`)}
			imageSize="small"
		/>
	)

	_navigateToSettings = () => {
		this.props.navigation.navigate("Settings")
	}

	_renderLessons = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<LessonsLoader />
				</View>
			)
		}

		return (
			<Fragment>
				<Text style={styles.rectTitle} testID="schedule">
					{strings("teacher.home.current_lesson")}
				</Text>
				{this.renderLesson(this.state.currentLesson)}
				<Separator />
				<Text style={styles.rectTitle}>
					{strings("teacher.home.next_lesson")}
				</Text>
				{this.renderLesson(this.state.nextLesson)}
			</Fragment>
		)
	}

	_renderPayments = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<PaymentsLoader />
				</View>
			)
		}
		let sumColor = colors.green
		if (this.state.sum < 0) sumColor = "red"
		let list
		if (this.state.payments.length == 0) {
			list = this._renderEmpty("payments")
		} else {
			list = (
				<FlatList
					data={this.state.payments.slice(0, 2)}
					renderItem={this.renderPaymentItem}
					keyExtractor={item => `payment${item.id}`}
				/>
			)
		}
		return (
			<Fragment>
				<View style={styles.amountView}>
					<Text style={{ ...styles.amount, color: sumColor }}>
						{this.state.sum}₪
					</Text>
					<TouchableOpacity
						onPress={() =>
							this.props.navigation.navigate("AddPayment")
						}
					>
						<Text style={styles.addPayment}>
							{strings("teacher.home.add_payment")}
						</Text>
					</TouchableOpacity>
				</View>
				<Separator />
				{list}
			</Fragment>
		)
	}

	render() {
		return (
			<ScrollView>
				<View style={styles.container}>
					<View style={styles.settingsIcon}>
						<TouchableOpacity
							onPress={this._navigateToSettings.bind(this)}
						>
							<Icon name="settings" type="material" size={24} />
						</TouchableOpacity>
					</View>
					<View testID="welcomeHeader" style={styles.welcomeHeader}>
						<UploadProfileImage
							style={styles.profilePic}
							image={getUserImage(this.props.user)}
							upload={async source => {
								await this.props.dispatch(
									uploadUserImage(source)
								)
							}}
						/>
						<Text style={styles.welcomeText}>
							{strings("teacher.home.welcome", {
								name: this.props.user.name.slice(0, NAME_LENGTH)
							})}
						</Text>
					</View>
					<ShadowRect style={styles.schedule}>
						{this._renderLessons()}
					</ShadowRect>

					<View style={styles.fullScheduleView}>
						<TouchableHighlight
							underlayColor="lightgray"
							onPress={() => {
								this.props.navigation.navigate("Schedule")
							}}
						>
							<Text style={styles.fullSchedule}>
								{strings("teacher.home.full_schedule")}
							</Text>
						</TouchableHighlight>
						<Icon
							size={20}
							color="rgb(12, 116, 244)"
							name="ios-arrow-dropleft-circle"
							type="ionicon"
						/>
					</View>
					<ShadowRect>
						<View style={{ flex: 1, flexDirection: "row" }}>
							<Text
								testID="monthlyAmount"
								style={styles.rectTitle}
							>
								{strings("teacher.home.monthly_amount")}
							</Text>
							<TouchableOpacity
								style={{
									flex: 1,
									alignItems: "flex-end",
									marginRight: "auto"
								}}
								onPress={() =>
									this.props.navigation.navigate({
										routeName: "Notifications",
										params: {},
										action: NavigationActions.navigate({
											routeName: "Main",
											params: {
												filter: "lessons/payments"
											}
										})
									})
								}
							>
								<Icon
									name="arrow-back"
									type="material"
									size={20}
								/>
							</TouchableOpacity>
						</View>
						{this._renderPayments()}
					</ShadowRect>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: 20,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	schedule: { minHeight: 220 },
	welcomeHeader: {
		alignSelf: "center",
		alignItems: "center",
		marginBottom: 20
	},
	settingsIcon: {
		position: "absolute",
		right: MAIN_PADDING
	},
	welcomeText: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	profilePic: {
		width: 44,
		height: 44,
		borderRadius: 22,
		marginBottom: 16
	},
	rectTitle: {
		fontSize: 16,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start"
	},
	lessonRow: {
		marginTop: 12
	},
	hours: { marginTop: 8 },
	fullScheduleView: {
		flexDirection: "row",
		flex: 1,
		alignSelf: "center",
		alignItems: "center",
		marginTop: 4,
		marginBottom: 24
	},
	fullSchedule: {
		color: "rgb(12, 116, 244)",
		marginTop: -4,
		fontWeight: "bold",
		marginRight: 8
	},
	places: {
		fontSize: 14,
		color: "gray"
	},
	placesImage: { marginTop: 8 },
	amountView: {
		marginTop: 16,
		alignSelf: "center"
	},
	amount: {
		fontFamily: "Assistant-Light",
		fontSize: 44
	},
	addPayment: {
		color: "rgb(12, 116, 244)",
		fontWeight: "bold",
		marginTop: 16,
		alignSelf: "center"
	},
	amountOfStudent: {
		marginTop: 4
	},
	nameStyle: {
		marginTop: 4,
		marginLeft: -2
	},
	paymentRow: {
		marginTop: 12
	},
	listLoader: { marginTop: 20, alignSelf: "center" },
	noLesson: {
		fontSize: 20,
		alignSelf: "center"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Home)
