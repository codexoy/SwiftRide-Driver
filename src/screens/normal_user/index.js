import React, { Component } from "react"
import { createBottomTabNavigator } from "react-navigation"
import Home from "./Home"
import { tabBarOptions } from "../consts"

export default createBottomTabNavigator(
	{
		Home: Home
	},
	tabBarOptions
)
