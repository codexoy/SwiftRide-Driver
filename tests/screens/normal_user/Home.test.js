import "react-native"
import React from "react"
import FetchService from "../../../src/services/Fetch"
import renderer from "react-test-renderer"

import { Home } from "../../../src/screens/normal_user/Home"

const fetchService = new FetchService()
jest.useFakeTimers()
describe("Home", () => {
	test("view renders correctly for teacher", () => {
		const user = { name: "test", teacher_id: 1 }

		const tree = renderer
			.create(<Home user={user} fetchService={fetchService} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
	test("view renders correctly for student", () => {
		const user = {
			name: "test2",
			my_teacher: { user: { name: "teacher" }, teacher_id: 1 }
		}
		const tree = renderer
			.create(<Home user={user} fetchService={fetchService} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
