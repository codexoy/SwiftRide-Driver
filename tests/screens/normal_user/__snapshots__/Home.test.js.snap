// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view renders correctly for student 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 26,
          "paddingRight": 26,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 120,
              "width": 120,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/images/pending.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginTop": 24,
          }
        }
      >
        ממתינים לאישורך!
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "marginTop": 12,
            "textAlign": "center",
          }
        }
      >
        נתוניך נקלטו במערכת, יש להמתין לאישורך על ידי המורה שאליו נרשמת.
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(12,116,244)",
            "borderRadius": 28,
            "elevation": 8,
            "height": 56,
            "justifyContent": "center",
            "marginTop": 40,
            "opacity": 1,
            "shadowColor": "rgb(12,116,244)",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 16,
            "width": 160,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          התנתק
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Home view renders correctly for teacher 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 26,
          "paddingRight": 26,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 120,
              "width": 120,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/images/pending.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginTop": 24,
          }
        }
      >
        ממתינים לאישורך!
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "marginTop": 12,
            "textAlign": "center",
          }
        }
      >
        נתוניך נקלטו במערכת, כעת יש להמתין לאישור המנהלים.
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(12,116,244)",
            "borderRadius": 28,
            "elevation": 8,
            "height": 56,
            "justifyContent": "center",
            "marginTop": 40,
            "opacity": 1,
            "shadowColor": "rgb(12,116,244)",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 16,
            "width": 160,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          התנתק
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
